docs/TemplateManagementCLI.md

```diff
@@ -0,0 +1,100 @@
+# Using Template Management CLI
+
+The command-line tool can be used to pull and push a template collection through remote registry (Now we only support Azure Container Registry). 
+
+## Prerequisites
+* Azure container registry - Create a container registry in your Azure subscription. For example, use the [Azure portal](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-portal) or the [Azure CLI](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-azure-cli).
+
+* Azure Active Directory service principal (optional) - If using service principal's identity for authentication, you need to create a [service principal](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal) to access your registry. Ensure that the service principal is assigned a role such as AcrPush so that it has permissions to push and pull artifacts.
+
+* Azure CLI (optional) - To use an individual identity, you need a local installation of the Azure CLI. Version 2.0.71 or later is recommended. Run `az --version` to find the version. If you need to install or upgrade, see [Install Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
+
+* Docker (optional) - To use an individual identity, you must also have Docker installed locally, to authenticate with the registry. Docker provides packages that easily configure Docker on any [macOS](https://docs.docker.com/docker-for-mac/), [Windows](https://docs.docker.com/docker-for-windows/), or [Linux](https://docs.docker.com/engine/install/) system.
+## Authentication
+
+Before pull & push operations, azure authentication is required for private registries. Customers can directly use individual login with Azure AD through [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli) or use identity (individual identity or Azure AD [service principal identity](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal)) to sign in the registry. 
+
+### Login Using Azure CLI
+
+After signing in to the Azure CLI with your identity, use the Azure CLI command `az acr login` to access the registry.
+```
+> az acr login --name <registry>
+```
+
+### Login Using Identity (individual or service principal indentity)
+
+* Docker login
+
+```
+> docker login <registry> -u <username> -p <password>
+```
+* Oras Login
+
+The [oras](https://github.com/deislabs/oras) tool oras.exe is packed in our repo, users can directly use it for login as follows.
+
+```
+>.\oras.exe login <registry> -u <username> -p <password>
+```
+
+## Push
+To push a template collection, the command is: 
+
+```
+push <ImageReference> InputTemplateFolder [ -n | --NewBaseLayer]
+```
```

Comment:
Is there a way to provide comments while pushing? A user may use these comments to differentiate between different template sets.

Reply:
For now, we couldn't add comments since there's no interface to edit the describe message for image on acr.

Reply:
possible to add it into metadata.json? I feel this feature will be quite useful.

Reply:
Yes, we can add it into metadata. 
But user can not read it unless pulling the image.